docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+# Contributing to LoopBack
+
+Contributions to Node.js include code, documentation, answering user questions, and advocating for all types of Node.js users. See our official documentation on loopback.io for more information common to all of our GitHub repositories:
+
+  - http://loopback.io/doc/en/contrib/index.html
+
+The rest of this document described technical details specific to loopack-next monorepo.
+
+## Setting up development environment
+
+We recommend our contributors to use VisualStudio Code with the following extensions installed:
+
+ - [Prettier - Code Formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) for automatic formatting of source files on save.
+
+ - [TSLint](https://marketplace.visualstudio.com/items?itemName=eg2.tslint) to highlight and auto-fix linting problems directly in the editor.
+
+Having said that, our project provides `package.json` scripts that are independent of any IDE, you are welcome to use your favorite IDE or editor instead of VS Code.
+
+## Common tasks
+
+### Build-test-lint cycle
+
+(to be described)
+
+### Run CI build locally
+
+(to be described)
+
+### Update project after switching git branches
+
+(to be described)
+
+## Improving project infrastructure
+
+Whenever making changes to our configuration of TypeScript compiler, TSLint or Prettier, extra care must be taken to avoid regressions, especially in the integration with VisualStudio Code.
+
+### Verify TypeScript setup
+
+#### Compilation errors
+
+1. Open any existing TypeScript file, e.g. `packages/src/index.ts`
+
+2. Add a small bit of code to break TypeScript's type checks, for example:
+
+    ```ts
+    const foo: number = 'bar';
+    ```
+
+3. Verify that VS Code editor has marked `foo` with a red underscore. Hover over `foo` and check the problem message. It should mention `[ts]` source, e.g.
+
+    ```text
+    [ts] Type '"bar"' is not assignable to type 'number'.
+
+    ```
+
+4. Check VS Code's [PROBLEMS Window](https://code.visualstudio.com/docs/getstarted/tips-and-tricks#_errors-and-warnings).  There should be an entry showing the same tslint error. When you click on the entry, it should jump on the problematic line.
+
+5. Close the editor tab. (This will clear the PROBLEMS entry reported by TSLint extension).
+
+6. Run the test task ("Tasks: Run test task"). This will invoke package scripts like `npm test` under the hood.
+
+7. Open "Tasks" OUTPUT window and verify that compilation error was parsed by VSCode.
+
+8. Verify that compilation errors are correctly associated with the problematic source code line.
+   _(2018-02-82: this is does not work now, `tsc` is reporting paths relative to individual package directories.)_
```

Comment:
FWIW: this used to work before, I suspect https://github.com/strongloop/loopback-next/pull/682 broke it.

The trick is that we need to run `tsc` from the root monorepo directory. 

https://github.com/strongloop/loopback-next/pull/682/files#diff-272a4137cf0e755785f43213ae0bae7fL14
> Under the hood, we change the current directory to loopback-next root
> and call tsc with project configuration from the original package's
> tsconfig.build.json. This way tsc error messages contain a path
> that's relative to loopback-next project root.

https://github.com/strongloop/loopback-next/pull/682/files#diff-272a4137cf0e755785f43213ae0bae7fL60

```
'-p',
// It's important to keep this path relative to rootDir
path.join(packageDir, 'tsconfig.build.json'),
```

I have some ideas how to fix this (basically introduce a new `lb-tsc` argument like `-r ../..` that will control the working directory of `tsc`), but I prefer to leave such changes out of this documentation-only pull request.